### PR TITLE
dataman: add read and write perf counters

### PR DESCRIPTION
**Pull request by dagar**
Adding read and write perf counters in dataman for the dm_read and dm_write operations. It's helpful to be able to keep an eye on these things in real operation.

Related: https://github.com/PX4/Firmware/issues/13536

### CODE REVIEW

File: src/modules/dataman/dataman.cpp [hunk 1]
```diff
@@ -1098,13 +1092,16 @@ dm_write(dm_item_t item, unsigned index, dm_persitence_t persistence, const void
 	work->write_params.count = count;
 
 	/* Enqueue the item on the work queue and wait for the worker thread to complete processing it */
-	return (ssize_t)enqueue_work_item_and_wait_for_result(work);
+	ssize_t ret = (ssize_t)enqueue_work_item_and_wait_for_result(work);
+	perf_end(_dm_write_perf);
```
Comment (bkueng) [inline src/modules/dataman/dataman.cpp]: There are more return's in between, where perf_end is not called (same below).

File: src/modules/dataman/dataman.cpp [hunk 2]
```diff
@@ -1098,13 +1092,16 @@ dm_write(dm_item_t item, unsigned index, dm_persitence_t persistence, const void
...
+	return ret;
 }
 
 /** Retrieve from the data manager file */
 __EXPORT ssize_t
 dm_read(dm_item_t item, unsigned index, void *buf, size_t count)
 {
+	perf_begin(_dm_read_perf);
```
Comment (bkueng) [inline src/modules/dataman/dataman.cpp]: This is invalid if `!is_running()`. Can you move it down?

File: src/modules/dataman/dataman.cpp [hunk 3]
```diff
@@ -1432,6 +1434,8 @@ task_main(int argc, char *argv[])
 	px4_sem_destroy(&g_work_queued_sema);
 	px4_sem_destroy(&g_sys_state_mutex_mission);
 	px4_sem_destroy(&g_sys_state_mutex_fence);
+	perf_free(_dm_read_perf);
+	perf_free(_dm_write_perf);
```
Comment (bkueng) [inline src/modules/dataman/dataman.cpp]: Can you add `_dm_write_perf = nullptr;`?

Comment (dagar) [reply src/modules/dataman/dataman.cpp]: Done.

File: src/modules/dataman/dataman.cpp (hunk 2 above)
Comment (dagar) [reply src/modules/dataman/dataman.cpp]: Done.

File: src/modules/dataman/dataman.cpp (hunk 1 above)
Comment (dagar) [reply src/modules/dataman/dataman.cpp]: Fixed.

**Comment by dagar** [comment]
> I'm surprised that we didn't have them already.

Likewise and why I tried to hastily add them. :smirk:

**Comment by dagar** [comment]
From SITL.
image

**Pull request merged**